### PR TITLE
70 dockerize keycloak ssi deployment project

### PR DESCRIPTION
**Pull request by forkimenjeckayang**

### CODE REVIEW

File: 0.start-kc-oid4vci.sh [hunk 1]
```diff
@@ -44,22 +44,6 @@ else
...
-        keycloak_pid=$(pgrep -f keycloak)
-        if [ -n "$keycloak_pid" ]; then
-            echo "Keycloak instance found (PID: $keycloak_pid) on $OS. Shutting it down..."
-            kill $keycloak_pid
-        fi
-        ;;
-    *)
-        echo "This script supports only Linux or macOS."
-        ;;
-esac
```
[the remark this answers is not in the document]
Comment (forkimenjeckayang) [reply 0.start-kc-oid4vci.sh:61]: Noted Sir 
Will create a new script and use

File: Dockerfile [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
-# Clone the repository
-RUN git clone --depth 1 https://github.com/keycloak/keycloak.git /tmp/keycloak
-WORKDIR /tmp/keycloak
+# Set the working directory
+WORKDIR /opt/keycloak/
 
-# Build Keycloak without running tests
-RUN mvn clean install -DskipTests
+# Install Git, apt-utils and other dependencies
+RUN apt-get update && apt-get install -y git apt-utils
```
[the remark this answers is not in the document]
Comment (forkimenjeckayang) [reply Dockerfile:23]: Had issues before when we using just one in the build and runtime phase but will look into again with the new script i will create

File: Dockerfile [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
-# Build Keycloak with the oid4vc-vci feature
-RUN bin/kc.sh build --features="oid4vc-vci"
+# Copy the environment variable file from the build stage
+COPY --from=builder /app/.env /opt/keycloak/
 
-# Entry point
-ENTRYPOINT ["bin/kc.sh", "start-dev", "--features=oid4vc-vci"]
\ No newline at end of file
+# Set the entry point
+ENTRYPOINT ["sh", "-c", "set -a && . /opt/keycloak/.env && set +a && cd \"$KC_INSTALL_DIR\" && exec bin/kc.sh \"$KC_START\" --features=oid4vc-vci"]
```
Comment (IngridPuppet) [inline Dockerfile:29]: Have you tried running the container with `KC_START` quoted? I have an error that actually makes sense:

```
Option: 'start --hostname-strict' is not expected to contain whitespace, please remove any unnecessary quoting/escaping
Did you mean: kc.sh start or kc.sh start-dev or kc.sh import?
Try 'kc.sh --help' for more information on the available options.
```

```suggestion
ENTRYPOINT ["sh", "-c", "set -a && . /opt/keycloak/.env && set +a && cd \"$KC_INSTALL_DIR\" && exec bin/kc.sh $KC_START --features=oid4vc-vci"]
```

File: build-kc-oid4vci.sh
Comment (IngridPuppet) [inline build-kc-oid4vci.sh:1]: You should reuse this script in `./0.start-kc-oid4vci.sh` to avoid code duplication.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

**Review by IngridPuppet — CHANGES_REQUESTED** [review]
Can you check again ? I can't run the docker image after build.
```
sh: 1: /opt/keycloak/target/tools/keycloak-999.0.0-SNAPSHOT: Permission denied
```

**Comment by forkimenjeckayang** [comment]
It should work now Sir
Saw what caused the error

**Review by IngridPuppet — APPROVED** [review]

**Pull request merged**